File: src/TypeScript.Modeller.Tests.Integration/SampleTwoExpected/ProfileViewModel.ts
import { CarViewModel } from './CarViewModel';
import { UnmarkedViewModel } from './UnmarkedViewModel';
import { AddressViewModel } from './AddressViewModel';

export class ProfileViewModel {
	public applicantId?: string = undefined;
	public firstName: string = '';
	public middleName: string = '';
	public lastName: string = '';
	public dateOfBirth?: Date = undefined;
	public gender: string = '';
	public email: string = '';
	public mobileNumber: string = '';
	public applicantStatus: number = 0;
	public car: CarViewModel | undefined = undefined;
	public unmarked: UnmarkedViewModel | undefined = undefined;
	public addresses: AddressViewModel[] = [];
}

File: src/Sample.Assembly.Two/ProfileViewModel.cs
using System;
using System.Collections.Generic;
using Sample.Assembly.One;
using Typescript.Modeller;

namespace Sample.Assembly.Two
{
    [TypeScriptViewModel]
    public class ProfileViewModel
    {
        public Guid? ApplicantId { get; set; }

        public string FirstName { get; set; }

        public string MiddleName { get; set; }

        public string LastName { get; set; }

        public DateTime? DateOfBirth { get; set; }

        public string Gender { get; set; }

        public string Email { get; set; }

        public string MobileNumber { get; set; }

        public virtual int ApplicantStatus { get; set; }

        public virtual CarViewModel Car { get; set; }

        public virtual UnmarkedViewModel Unmarked { get; set; }

        public virtual ICollection<AddressViewModel> Addresses { get; set; }
    }
}